backend/src/main/java/com/skr/virtuallibrary/services/BookListService.java

```diff
@@ -0,0 +1,147 @@
+                throw new BookNotFoundException("Book not found with id: " + bookId);
+            }
+        }
+
+        return modelMapper.toBookListDto(bookListRepository.save(
+                BookList.builder()
+                        .userId(user.getId())
+                        .name(bookListDto.getName())
+                        .deletable(true)
+                        .bookIds(bookListDto.getBookIds())
+                        .build()
+        ));
+    }
+
+    public BookListDto addBookToList(String id, String bookId) {
+        User user = userService.getCurrentUser();
+        BookList bookList = bookListRepository.findById(id)
+                .orElseThrow(() -> new BookListNotFoundException(BOOK_LIST_NOT_FOUND + id));
+        if (!bookList.getUserId().equals(user.getId())) {
+            throw new AccessForbiddenException(ACCESS_DENIED);
+        }
+        if (bookList.getBookIds().contains(bookId)) {
+            return modelMapper.toBookListDto(bookList);
+        }
+        if (bookRepository.findById(bookId).isEmpty()) {
+            throw new BookNotFoundException("Book not found with id: " + bookId);
+        }
+
+        bookList.getBookIds().add(bookId);
+        return modelMapper.toBookListDto(bookListRepository.save(bookList));
+    }
+
+    public BookListDto removeBookFromList(String id, String bookId) {
+        User user = userService.getCurrentUser();
+        BookList bookList = bookListRepository.findById(id)
+                .orElseThrow(() -> new BookListNotFoundException(BOOK_LIST_NOT_FOUND + id));
+        if (!bookList.getUserId().equals(user.getId())) {
+            throw new AccessForbiddenException(ACCESS_DENIED);
+        }
+        bookList.getBookIds().remove(bookId);
+        return modelMapper.toBookListDto(bookListRepository.save(bookList));
+    }
+
+    public BookListDto changeName(String id, String name) {
+        User user = userService.getCurrentUser();
+        BookList bookList = bookListRepository.findById(id)
+                .orElseThrow(() -> new BookListNotFoundException(BOOK_LIST_NOT_FOUND + id));
+        if (!bookList.getUserId().equals(user.getId())) {
+            throw new AccessForbiddenException(ACCESS_DENIED);
+        }
+        if (!bookListRepository.findAllByNameAndUserId(name, user.getId()).isEmpty()) {
+            throw new BookListAlreadyExistsException(bookList.getName());
+        }
+
+        bookList.setName(name);
+        return modelMapper.toBookListDto(bookListRepository.save(bookList));
+    }
+
+    public void deleteBookList(String id) {
+        User user = userService.getCurrentUser();
+        BookList bookList = bookListRepository.findById(id)
+                .orElseThrow(() -> new BookListNotFoundException(BOOK_LIST_NOT_FOUND + id));
+        if (!bookList.getUserId().equals(user.getId())) {
+            throw new AccessForbiddenException(ACCESS_DENIED);
+        }
+        if (!bookList.isDeletable()) {
```

Comment:
Nie wiem, czy ta lista której nie można usunąć to dobry pomysł, bo trochę komplikuje kod i wydaje mi się jednak że użytkownik powinien móc ją usunąć jeśli nie chce jej mieć u siebie.

Reply:
No pytanie jak to robimy: możemy tą listę potraktować jak playlistę *Do obejrzenia* na YT; wtedy taka lista może mieć oddzielny przycisk na frontendzie (*dodaj do listy ...* i *przeczytaj później* (czy jakoś tak to nazwać) oddzielnie )